lib/mix/lib/mix/sync/pubsub.ex

```diff
@@ -127,6 +127,7 @@ defmodule Mix.Sync.PubSub do
   @impl true
   def init({}) do
     state = %{port: nil, hash_to_pids: %{}}
+    init_base_path()
```

Comment:
Instead of init, let's mkdir_p it when needed, same as above. This is just to be safe, in case the tmp dir is removed at some point as the elixir process keeps running.